src/photon_producer.cpp

```diff
@@ -4,86 +4,115 @@
+  pico.out_nphoton() = 0;
+  // pico.out_nfsrphoton() = 0;
   vector<int> sig_photon_nano_idx;
   int nphotons(0), ndr(0), shift(0);
+
+  for(int iph(0); iph < nano.nFsrPhoton(); ++iph){
+    if (nano.FsrPhoton_pt()[iph] <= FsrPhotonPtCut) continue;
+    if (fabs(nano.FsrPhoton_eta()[iph]) > FsrPhotonEtaCut) continue;
+    if (nano.FsrPhoton_relIso03()[iph] > FsrPhotonIsoCut) continue;
+    if (nano.FsrPhoton_dROverEt2()[iph] > FsrPhotondRCut) continue;
+    pico.out_fsrphoton_pt().push_back(nano.FsrPhoton_pt()[iph]);
+    pico.out_fsrphoton_eta().push_back(nano.FsrPhoton_eta()[iph]);
+    pico.out_fsrphoton_phi().push_back(nano.FsrPhoton_phi()[iph]);
+    pico.out_fsrphoton_reliso().push_back(nano.FsrPhoton_relIso03()[iph]);
+    pico.out_fsrphoton_muonidx().push_back(nano.FsrPhoton_muonIdx()[iph]);
+    pico.out_fsrphoton_droveret2().push_back(nano.FsrPhoton_dROverEt2()[iph]);
+    pico.out_nfsrphoton()++;
+  }
+
   for(int iph(0); iph<nano.nPhoton(); ++iph){
-    if (nano.Photon_pt()[iph] <= PhotonPtCut) continue;
-    if (fabs(nano.Photon_eta()[iph]) > PhotonEtaCut) continue;
+    float pt = nano.Photon_pt()[iph];
+    float eta = nano.Photon_eta()[iph];
+    float phi = nano.Photon_phi()[iph];
+    float mva = nano.Photon_mvaID()[iph];
+    bool eVeto = nano.Photon_electronVeto()[iph];
+
+    if (pt <= PhotonPtCut) continue;
+    if (fabs(eta) > PhotonEtaCut) continue;
     if (!(nano.Photon_isScEtaEB()[iph] || nano.Photon_isScEtaEE()[iph])) continue;
-    // if (nano.Photon_pfRelIso03_all()[iph]==PhotonRelIsoCut) continue; // no isolation cut in 2016...?
-    bool isSig = nano.Photon_mvaID()[iph] > 0.2 && 
-                 nano.Photon_electronVeto()[iph] && 
-                 nano.Photon_pt()[iph] > SignalPhotonPtCut;
+
     // Find min(dR) between photon and signal lepton
     double minLepDR(999.);
     for(size_t iel(0); iel<sig_el_nano_idx.size(); iel++) {
-      double tempDR = dR(nano.Photon_eta()[iph], nano.Electron_eta()[sig_el_nano_idx.at(iel)], 
-                         nano.Photon_phi()[iph], nano.Electron_phi()[sig_el_nano_idx.at(iel)]);
+      double tempDR = dR(eta, nano.Electron_eta()[sig_el_nano_idx.at(iel)],
+                         phi, nano.Electron_phi()[sig_el_nano_idx.at(iel)]);
       if(tempDR < minLepDR) minLepDR = tempDR;
     }
     for(size_t imu(0); imu<sig_mu_nano_idx.size(); imu++) {
-      double tempDR = dR(nano.Photon_eta()[iph], nano.Muon_eta()[sig_mu_nano_idx.at(imu)], 
-                         nano.Photon_phi()[iph], nano.Muon_phi()[sig_mu_nano_idx.at(imu)]);
+      double tempDR = dR(eta, nano.Muon_eta()[sig_mu_nano_idx.at(imu)],
+                         phi, nano.Muon_phi()[sig_mu_nano_idx.at(imu)]);
       if(tempDR < minLepDR) minLepDR = tempDR;
     }
+
+    bool isSig = (((fabs(eta) < 1.4442 && mva > -0.4) ||
+		  (fabs(eta) > 1.566 && fabs(eta) < 2.5 && mva > -0.58)) &&
+		  eVeto && minLepDR > 0.4 && 
```

Comment:
Are we sure that we want to remove the minLepDR < 0.4 from the signal photon collection?

Reply:
Until we finalize a baseline selection we will have to stick with the selections from the Run II analysis.

Reply:
But we can still leave photons with minLepDR < 0.4 to be reconstructed into ZGamma candidates?

Reply:
We can still reconstruct the ZGamma candidates with minLepDR < 0.4 at the draw_pico level.